lib/grammar.js

```diff
@@ -0,0 +1,328 @@
+'use strict';
+
+/**
+ * Removes `[`, `]` and any trailing quantifiers from the source of a RegExp
+ * @param {RegExp} regexp
+ */
+function chars(regexp) {
+	if (regexp.source[0] !== '[') {
+		throw new Error(regexp + ' can not be used with chars');
+	}
+	return regexp.source.slice(1, regexp.source.lastIndexOf(']'));
+}
+
+/**
+ * Creates a new character list regular expression,
+ * by removing `search` from the source of `regexp`.
+ * @param {RegExp} regexp
+ * @param {string} search the characters to remove
+ * @returns {RegExp & {chars:string}}
+ */
+function chars_without(regexp, search) {
+	if (regexp.source[0] !== '[') {
+		throw new Error('/' + regexp.source + '/ can not be used with chars_without');
+	}
+	if (!search) {
+		throw new Error(search + ' is falsy');
+	}
+	if (regexp.source.indexOf(search) === -1) {
+		throw new Error('"' + search + '" is not is /' + regexp.source + '/');
+	}
+	var result = new RegExp(regexp.source.replace(search, ''));
+	result.chars = chars(result);
+	return result;
+}
```

Comment:
This function should be marked as unsafe, or add some warnings.
For example, if you remove `-`, it will remove it from the first range it encounters, if the literal `-` is not present.

Reply:
I already have it in my local typescript version (only accepting `-` if it is the first char in the character class), will add that.

Reply:
> I already have it in my local typescript version (only accepting `-` if it is the first char in the character class), will add that.

That is just one of the issues.
For example:
```
var test = /[0-9]/
var test_without = chars_without(test, '0')
test_without.source == [-9]
```